Python/errors.c

```diff
@@ -151,7 +151,14 @@ _PyErr_SetObject(PyThreadState *tstate, PyObject *exception, PyObject *value)
     }
     Py_XINCREF(value);
     /* Normalize the exception */
-    if (value == NULL || (PyObject *)Py_TYPE(value) != exception) {
+    int is_subclass = 0;
+    if (value != NULL && PyExceptionInstance_Check(value)) {
+        is_subclass = PyObject_IsSubclass((PyObject *)Py_TYPE(value), exception);
+        if (is_subclass < 0) {
+            return;
```

Comment:
I think this would leak the reference to value (it was INCREFed on line 152).

Reply:
I see what you mean, but this should have caused 

`./python.exe -m test -v -R3:3 test_capi -m test_set_object`

to fail and it didn't, so I want to understand why.

Reply:
Did you add a test that involves an __issubclasscheck__ that raises an exception?

Reply:
ah that's the case where the is subclass check fails. it's not covered in the tests.